### PR TITLE
AVRO-3678: [Rust] Support writing float value to field defined as double

### PR DESCRIPTION
**Pull request by shaeqahmed**
## What is the purpose of the change

This pull request adds support for validating/resolving a float value against a destination schema where the corresponding field is defined as a double, fixing AVRO-3678. This makes the behavior similar to how we currently support other lossless upcasting of numeric types, such as int -> long, which should be expected behavior.

https://issues.apache.org/jira/browse/AVRO-3678

## Verifying this change

This change added tests and can be verified as follows:
- Added tests that validates that a float value can be resolved & validated against a schema double field.

## Documentation

- Does this pull request introduce a new feature? **no**

### CODE REVIEW

**Comment by shaeqahmed** [comment]
@martin-g This is my second time opening a contribution to the Avro project, so please let me know if I should make any changes. Thanks!

**Comment by martin-g** [comment]
Please squash your commits and add `AVRO-3678: ` as a prefix for the commit message.
At the moment there are 14 commits and most of them are merge commits.

Also please address the clippy errors.

File: lang/rust/avro/src/types.rs [hunk 1]
```diff
@@ -1565,6 +1570,12 @@ Field with name '"b"' is not a member of the map items"#,
         assert!(value.resolve(&Schema::TimestampMicros).is_err());
     }
 
+    #[test]
+    fn resolve_float_to_double() {
```
Comment (martin-g) [inline lang/rust/avro/src/types.rs:1574]:
```suggestion
    fn avro_3678_resolve_float_to_double() {
```

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
@shaeqahmed Please let me know if you need help with git and/or clippy!

**Comment by martin-g** [comment]
Closing in favour of https://github.com/apache/avro/pull/1991.
It has clear commit history and solves the clippy errors.
Thank you, @shaeqahmed !

**Comment by shaeqahmed** [comment]
My bad for the delay, was travelling and got sick so didn't get the chance to address the comments till now. Looks like you fixed the minor issues and merged it, thank you!

**Pull request closed without merging**